pkg/tbtc/signing_test.go

```diff
@@ -202,6 +374,230 @@ func TestSigningRetryLoop(t *testing.T) {
+		broadcastingMembersIndexes []group.MemberIndex
+		expectedResults            map[group.MemberIndex][]group.MemberIndex
+	}{
+		"all members broadcasted announcements": {
+			message:                    big.NewInt(100),
+			broadcastingMembersIndexes: []group.MemberIndex{1, 2, 3, 4, 5},
+			expectedResults: map[group.MemberIndex][]group.MemberIndex{
+				1: {1, 2, 3, 4, 5},
+				2: {1, 2, 3, 4, 5},
+				3: {1, 2, 3, 4, 5},
+				4: {1, 2, 3, 4, 5},
+				5: {1, 2, 3, 4, 5},
+			},
+		},
+		"part of members broadcasted announcements": {
+			message:                    big.NewInt(200),
+			broadcastingMembersIndexes: []group.MemberIndex{1, 3, 5},
+			expectedResults: map[group.MemberIndex][]group.MemberIndex{
+				1: {1, 3, 5},
+				3: {1, 3, 5},
+				5: {1, 3, 5},
+			},
+		},
+	}
+
+	for testName, test := range tests {
+		t.Run(testName, func(t *testing.T) {
+			broadcastChannel, err := localProvider.BroadcastChannelFor(
+				test.message.Text(16),
+			)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			membershipValidator := group.NewMembershipValidator(
+				&testutils.MockLogger{},
+				signingGroupOperators,
+				localChain.Signing(),
+			)
+
+			announcer := newBroadcastSigningAnnouncer(
+				chainConfig,
+				broadcastChannel,
+				membershipValidator,
+			)
+
+			resultsChan := make(
+				chan *memberResult,
+				len(test.broadcastingMembersIndexes),
+			)
+			errorsChan := make(
+				chan *memberError,
+				len(test.broadcastingMembersIndexes),
+			)
+
+			wg := sync.WaitGroup{}
+			wg.Add(len(test.broadcastingMembersIndexes))
+
+			for _, broadcastingMemberIndex := range test.broadcastingMembersIndexes {
+				go func(memberIndex group.MemberIndex) {
+					defer wg.Done()
+
+					ctx, cancelCtx := context.WithTimeout(
+						context.Background(),
+						100*time.Millisecond,
+					)
+					defer cancelCtx()
+
+					readyMembersIndexes, err := announcer.announce(
+						ctx,
+						memberIndex,
+						test.message,
+						1,
+					)
+					if err != nil {
+						errorsChan <- &memberError{memberIndex, err}
+						return
+					}
+
+					resultsChan <- &memberResult{memberIndex, readyMembersIndexes}
+				}(broadcastingMemberIndex)
+			}
+
+			wg.Wait()
+
+			close(resultsChan)
+			results := make(map[group.MemberIndex][]group.MemberIndex)
+			for r := range resultsChan {
+				results[r.memberIndex] = r.readyMembersIndexes
+			}
+
+			close(errorsChan)
+			errors := make(map[group.MemberIndex]error)
+			for e := range errorsChan {
+				errors[e.memberIndex] = e.err
+			}
+
+			testutils.AssertIntsEqual(
+				t,
+				"errors count",
+				0,
+				len(errors),
+			)
+
+			if !reflect.DeepEqual(test.expectedResults, results) {
+				t.Errorf(
+					"unexpected results\n"+
+						"expected: [%v]\n"+
+						"actual:   [%v]",
+					test.expectedResults,
+					results,
+				)
+			}
+		})
+	}
+}
+
+type mockSigningAnnouncer struct {
+	// outgoingAnnouncements holds all announcements that are supposed to
+	// be sent by the announcer.
```

Comment:
```suggestion
	// outgoingAnnouncements holds all announcements that are 
	// sent by the announcer.
```

I am not sure if this is "supposed to" because it's set in `announce` when the announce happens. So I think this is "what is sent by the announcer" and we later assert it against what it was supposed to send.

Reply:
https://github.com/keep-network/keep-core/pull/3361/commits/9d61a51cd434b23fdb57c54df08f068592fdec7a